src/v8n.js

```diff
@@ -37,6 +39,33 @@ function proxyContext(context) {
   });
 }
 
+function proxyContextEs5(context) {
+  Object.keys(availableRules).forEach(prop => {
+    context[prop] = (...args) => {
+      const newContext = proxyContextEs5(context._clone());
+      return newContext._applyRule(availableRules[prop], prop)(...args);
+    };
+  });
+
+  Object.keys(customRules).forEach(prop => {
+    context[prop] = (...args) => {
+      const newContext = proxyContextEs5(context._clone());
+      return newContext._applyRule(customRules[prop], prop)(...args);
+    };
+  });
+
+  Object.keys(availableModifiers).forEach(prop => {
+    Object.defineProperty(context, prop, {
+      get: () => {
+        const newContext = proxyContextEs5(context._clone());
+        return newContext._applyModifier(availableModifiers[prop], prop);
+      }
+    });
+  });
+
+  return context;
```

Comment:
There is generally a lot of repetition in here. I get that it works and I appreciate that, but since we're going through the effort of building a "polyfill" of sorts I suggest it's done in a clean way.